package NBlender.NProto;

import "mapreduce/yt/interface/protos/extension.proto";

option go_package = "a.yandex-team.ru/kernel/blender/factor_storage/protos";
message TFactorStorages {
    option (NYT.default_field_flags) = SERIALIZATION_YT;

    message TStaticFactorStorage {
        option (NYT.default_field_flags) = SERIALIZATION_YT;
        option (NYT.default_field_flags) = OPTIONAL_LIST;

        repeated float Value = 1 [packed=true];
    }

    message TDynamicFactorGroup {
        option (NYT.default_field_flags) = SERIALIZATION_YT;
        option (NYT.default_field_flags) = OPTIONAL_LIST;

        repeated float Value = 1 [packed=true];
        optional string Key = 2;
    }

    message TDynamicFactorStorage {
        option (NYT.default_field_flags) = SERIALIZATION_YT;
        option (NYT.default_field_flags) = OPTIONAL_LIST;

        repeated TDynamicFactorGroup DynamicFactorGroup = 1;
    }
    optional TStaticFactorStorage StaticStorage = 1;
    optional TDynamicFactorStorage DynamicStorage = 2;
}
